### PR TITLE
Enforce charset in String to bytes conversion to support non ascii pl…

### PR DESCRIPTION
**Pull request by vaxvms**
…atform.

### CODE REVIEW

File: src/main/java/com/googlecode/jsendnsca/encryption/TripleDESEncryptor.java [hunk 1]
```diff
@@ -39,19 +39,19 @@
...
+            final IvParameterSpec iv = new IvParameterSpec(initVectorBytes);
 
-        try {
             final Cipher cipher = Cipher.getInstance(DES_TRANSFORMATION);
             cipher.init(ENCRYPT_MODE, key, iv);
             final byte[] cipherText = cipher.doFinal(passiveCheckBytes);
 
             System.arraycopy(cipherText, 0, passiveCheckBytes, 0, passiveCheckBytes.length);
-        } catch (GeneralSecurityException e) {
+        } catch (Exception e) {
```
Comment (rajneeshpatel) [inline src/main/java/com/googlecode/jsendnsca/encryption/TripleDESEncryptor.java]: Whats the reason for weakening from GeneralSecurityException to Exception?

File: src/main/java/com/googlecode/jsendnsca/encryption/XorEncryptor.java [hunk 2]
```diff
@@ -39,14 +39,18 @@ public void encrypt(byte[] passiveCheckBytes, byte[] initVector, String password
...
-                if (x >= passwordBytes.length) {
-                    x = 0;
+                for (int y = 0, x = 0; y < passiveCheckBytes.length; y++, x++) {
+                    if (x >= passwordBytes.length) {
+                        x = 0;
+                    }
+                    passiveCheckBytes[y] ^= passwordBytes[x];
                 }
-                passiveCheckBytes[y] ^= passwordBytes[x];
+            } catch (Exception e) {
```
Comment (rajneeshpatel) [inline src/main/java/com/googlecode/jsendnsca/encryption/XorEncryptor.java]: AFAICS, there was no checked exceptions being thrown so is there any need to catch runtime exceptions and throw them wrapped in RuntimeException?

File: src/main/java/com/googlecode/jsendnsca/utils/ByteArrayUtils.java [hunk 3]
```diff
@@ -95,7 +95,11 @@ public static void writeInteger(byte[] dest, int value, int offset) {
             value = value.substring(0, fixedSize);
         }
 
-        System.arraycopy(value.getBytes(), 0, myBytes, 0, value.getBytes().length);
+        try {
+            System.arraycopy(value.getBytes("US-ASCII"), 0, myBytes, 0, value.getBytes("US-ASCII").length);
+        } catch (Exception e) {
```
Comment (rajneeshpatel) [inline src/main/java/com/googlecode/jsendnsca/utils/ByteArrayUtils.java]: Same again, any need to catch and wrap as RuntimeException?

File: src/main/java/com/googlecode/jsendnsca/encryption/TripleDESEncryptor.java (hunk 1 above)
Comment (vaxvms) [reply src/main/java/com/googlecode/jsendnsca/encryption/TripleDESEncryptor.java]: charset is correctly handled in BlowfishEncryptor and AESEncryptor and they catch Exception.
To be consistent, i've used the same pattern.

File: src/main/java/com/googlecode/jsendnsca/encryption/XorEncryptor.java (hunk 2 above)
Comment (vaxvms) [reply src/main/java/com/googlecode/jsendnsca/encryption/XorEncryptor.java]: According to javadoc getBytes(String) method throws an UnsupportedEncodingException :

public byte[] getBytes(String charsetName)
                throws UnsupportedEncodingException

I've used the same pattern as in BlowfishEncryptor and AESEncryptor

**Comment by vaxvms** [comment]
Is there anything i have to change to get my PR accepted ?

**Comment by rajneeshpatel** [comment]
Merged, will double check exact exceptions when I get a chance

**Pull request merged**